reactionimp/icemachinesplus
# (fix) Set "submit for settlement" when capture_at_auth is true

Resolves #123  
Impact: **major**  
Type: **feature**

## Issue
When payments are submitted to Braintree, they are by default set to "authorized". I

## Solution
 Added a "Capture at Auth" setting that will submit the transaction for settlement


## Breaking changes IMPORTANT
This submits the payment for settlement and sets the payment method mode to "capture" it doesn't change it's status in the order dashboard because this doesn't fit our current workflow model so the order will still show up in the order dashboard. Based on my discussions with Vishal this doesn't seem to be a problems for Ice Machines as they are using Netsuite for all their Order management/Fullfilment.

The changes to get this into the workflow properly will require some changes to core and some more investigation so this is sort of a quick fix/workaround.

If this temporary solution is not sufficient, don't merge this pull request.


## Testing
1. Set "Capture at Auth" in the Braintree settings panel
1. Place an order (< $1k or > $4k)
1. Log into Braintree admin panel
1. Observe that rather than "Authorized" the transaction status is "Submitted for Settlement"
